lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -0,0 +1,171 @@
+    plugin_type: inventory
+    authors:
+      - Stefan Heitmüller (stefan.heitmueller@gmx.com)
+    short_description: Ansible dynamic inventory plugin for Docker swarm nodes.
+    requirements:
+        - python >= 2.7
+        - Docker SDK for Python > 1.10.0
+    description:
+        - Reads inventories from the Docker swarm API.
+        - Uses a YAML configuration file docker_swarm.[yml|yaml].
+    options:
+        plugin:
+            description: The name of this plugin, it should always be set to 'docker_swarm' for this plugin to recognize it as it's own.
+            type: str
+            required: true
+            choices: docker_swarm
+        host:
+            description: Socket of a Docker swarm manager node (tcp,unix).
+            type: str
+            required: true
+        verbose_output:
+            description: Toggle to (not) include all available nodes metadata (e.g. Platform, Architecture, OS, EngineVersion)
+            type: bool
+            choices: [yes, no]
+            default: yes
+        tls:
+            description: Connect using TLS without verifying the authenticity of the Docker host server.
+            type: bool
+            choices: [yes, no]
+            default: no
+        tls_verify:
+            description: Connect using TLS without verifying the authenticity of the Docker host server.
+            type: bool
+            choices: [yes, no]
+            default: no
+        key_path:
+            description: Path to the client's TLS key file.
+            type: path
+        cacert_path:
+            description: Use a CA certificate when performing server verification by providing the path to a CA certificate file.
+            type: path
+        cert_path:
+            description: Path to the client's TLS certificate file.
+            type: path
+        tls_hostname:
+            description: When verifying the authenticity of the Docker Host server, provide the expected name of the server.
+            type: str
+'''
+
+EXAMPLES = '''
+# docker_swarm.yml
+plugin: docker_swarm
+host: tcp://my-docker-host:2375
+'''
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.inventory.group import to_safe_group_name as original_safe
+from ansible.module_utils._text import to_native
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.release import __version__
+
+try:
+    import docker
+    HAS_DOCKER = True
+except ImportError:
+    HAS_DOCKER = False
+
+
+class InventoryModule(BaseInventoryPlugin):
+    ''' Host inventory parser for ansible using Docker swarm as source. '''
+
+    NAME = 'docker_swarm'
+
+    def _get_tls_config(self, **kwargs):
+        try:
+            tls_config = docker.tls.TLSConfig(**kwargs)
+            return tls_config
+        except Exception as e:
+            raise AnsibleError('Unable to setup TLS, this was the original exception: %s' % to_native(e))
+
+    def _get_tls_connect_params(self):
```

Comment:
```suggestion
    def _set_tls_connect_params(self):
```

This method actually set the final TLS connection parameters

Reply:
I disagree: it returns the TLS connection parameters object. So it should not have `set` in its name. It can be `get` or `create` or `retrieve` or whatever, but not `set`. It's not a setter function.